src/auto/AutoCriterion.lua

```diff
@@ -0,0 +1,26 @@
+local AutoCriterion, parent = torch.class('nn.AutoCriterion', 'nn.Criterion')
```

Comment:
needs a 

```lua
require 'nn'
```

Is it also good practice to put the nn.AutoCriterion module under the nn namespace? Maybe it is better to put it under the autograd namespace itself? like: autograd.Criterion